services/azure-devops/azure-devops-artifacts.tester.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+// const Joi = require('joi')
+const t = (module.exports = require('../tester').createServiceTester())
+const {
+  isVPlusTripleDottedVersion,
+  isVPlusDottedVersionNClausesWithOptionalSuffix,
+} = require('../test-validators')
+
+// https://dev.azure.com/dotnet/NuGetPackageExplorer/_packaging?_a=feed&feed=BuildPackages is a public Azure DevOps project
+// that this badge will also use for testing purposes
+
+t.create('release package')
+  .get(
+    '/v/dotnet/NuGetPackageExplorer/BuildPackages/NuGetPackageExplorer.Core.json'
+  )
+  .expectBadge({
+    label: 'Azure Artifacts NuGet',
+    message: isVPlusTripleDottedVersion,
+    color: 'blue',
+  })
+
+t.create('prerelease package')
+  .get(
+    '/vpre/dotnet/NuGetPackageExplorer/BuildPackages/Microsoft.Build.NuGetSdkResolver.json'
+  )
+  .expectBadge({
+    label: 'Azure Artifacts NuGet',
+    message: isVPlusDottedVersionNClausesWithOptionalSuffix,
+    color: 'yellow',
+  })
```

Comment:
We'll want to have some tests that exercise the other flows (such as a non-existent package, feed, project, etc.). 

I'm a bit unclear on the path where the requested package is found but the normalized name doesn't match the package name, but we'd ideally have a test for that case too (as noted above moving that logic to a `transform` function would likely make this easier to test, [here's an example of some other unit tests for service class functions](https://github.com/badges/shields/blob/master/services/codecov/codecov.spec.js))